### PR TITLE
refactor: split dataset registry into per-category YAML files

### LINKED ISSUE #31: Split dataset registry into per-category YAML files
## Summary

Split the single `configs/datasets.yml` into per-category files under `configs/datasets/` to keep individual files manageable as more datasets are registered.

## Motivation

- `nlcd_legacy` alone has 30 explicit download file entries (67 lines)
- With 8 planned categories and multiple datasets per category, a single file will become unwieldy
- Per-category files make it easier to review, edit, and extend individual categories

## Approach

1. **Enhance `load_registry()`** to auto-detect file vs directory — backward compatible with single-file mode
2. **Split into 8 category files** under `configs/datasets/`:
   - `topography.yml` — `dem_3dep_10m`
   - `soils.yml` — `polaris_100m`
   - `land_cover.yml` — `nlcd_legacy`, `nlcd_annual`
   - `hydrography.yml`, `geology.yml`, `climate.yml`, `snow.yml`, `water_bodies.yml` — placeholders (`datasets: {}`)
3. **Update `DEFAULT_REGISTRY`** from `Path("configs/datasets.yml")` to `Path("configs/datasets")`
4. **Add tests** for directory loading, duplicate detection, edge cases
5. **Update docs** (design.md reference, CLI `--registry` help text)

Each category file retains the `datasets:` root key, so any single file is independently valid as a registry.

## Files to modify

- `src/hydro_param/dataset_registry.py` — `load_registry()` + new helpers
- `src/hydro_param/pipeline.py` — `DEFAULT_REGISTRY`
- `tests/test_registry.py` — new tests + update real registry tests
- `configs/datasets/` — new directory with 8 files
- `configs/datasets.yml` — delete after split
- `docs/design.md` — update registry path reference

### PR DESCRIPTION
**Pull request by rmcd-mscb**
## Summary

- `load_registry()` now auto-detects file vs directory, with duplicate name detection and per-file Pydantic validation
- Split `configs/datasets.yml` into `configs/datasets/<category>.yml` (8 files: topography, soils, land_cover + 5 placeholders)
- `DEFAULT_REGISTRY` updated to `configs/datasets/`; single-file mode still works via `--registry`

Closes #31

## Test plan

- [x] 7 new directory-loading tests (merge, duplicates, empty dir, no datasets, non-registry files, nonexistent path, single-file compat)
- [x] 3 real registry integration tests updated to use directory path
- [x] All 128 tests pass, lint/mypy/pre-commit clean
- [x] Smoke test: `hydro-param datasets list` and `hydro-param datasets info nlcd_annual`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/hydro_param/dataset_registry.py [hunk 1]
```diff
@@ -211,17 +211,88 @@ def resolve_variable(
...
+    DatasetRegistry
+
+    Raises
+    ------
+    FileNotFoundError
+        If the directory contains no YAML files or no datasets.
+    ValueError
+        If a dataset name appears in more than one file.
+    """
+    yaml_files = sorted(directory.glob("*.yml")) + sorted(directory.glob("*.yaml"))
```
[the remark this answers is not in the document]
Comment (rmcd-mscb) [reply src/hydro_param/dataset_registry.py]: Fixed in 2c101dc — now sorts the combined list so ordering is purely alphabetical regardless of .yml vs .yaml extension.
